src/main/controllers/helpers/CitizenHubHelper.ts

```diff
@@ -81,3 +81,17 @@ export const userCaseContainsGeneralCorrespondence = (notifications: SendNotific
     notifications.some(it => it.value.sendNotificationSubject.includes(NotificationSubjects.GENERAL_CORRESPONDENCE))
   );
 };
+
+export const checkIfRespondentIsSystemUser = (userCase: CaseWithId): boolean => {
+  const repCollection = userCase.representatives;
+  const respondentCollection = userCase.respondents;
+
+  return (
+    respondentCollection !== undefined &&
+    repCollection !== undefined &&
+    respondentCollection.every(res => {
+      return repCollection.some(rep => res.ccdId === rep.respondentId);
+    }) &&
+    !repCollection.some(r => r.hasMyHMCTSAccount === YesOrNo.NO || r.hasMyHMCTSAccount === undefined)
+  );
```

Comment:
This seems very complex, is there a way we can break up the logic to make it more readable? Perhaps we can exit early from this if either respondentCollection or repCollection is undefined? 

```
if (!repCollection || !respondentCollection) {
  return false
}
```